File: parth/Week1/ValidPalindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        for(int i=0, j=s.size()-1;i<j;)
        {
            if(!(isalpha(s[i]) || isdigit(s[i])))
            {
                ++i;
                continue;
            }
            if(!(isalpha(s[j]) || isdigit(s[j])))
            {
                --j;
                continue;
            }
            if (tolower(s[i]) != tolower(s[j]))
            {
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
};

File: Manali/Week1/ValidPalindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if len(s)==1: return True
        start = 0
        end = len(s)-1
        while start<end :
            if (not s[start].isalpha()) and (not s[start].isdigit()):
                start+=1
                continue
            if (not s[end].isalpha()) and (not s[end].isdigit()):
                end-=1
                continue
            if s[start].lower()==s[end].lower():
                start+=1
                end-=1
            else: return False
        return True
        